manifests/server/index.pp

```diff
@@ -0,0 +1,35 @@
+define postgresql::server::index (
+  Enum['absent','present'] $ensure,
+  String        $db,
+  String        $index_name,
+  String        $table_name,
+  Array[String] $column_names,
+  String        $psql_user         = $postgresql::server::user,
+  String        $psql_group        = $postgresql::server::group,
+  Integer       $port              = $postgresql::server::port,
+  Hash          $connect_settings  = $postgresql::server::default_connect_settings,
+  String        $psql_path         = $postgresql::server::psql_path
```

Comment:
`concurrently`, `unique`, `index_type`, `predicate` options would be nice.

https://www.postgresql.org/docs/9.5/static/sql-createindex.html